blosc/blosc2.c

```diff
@@ -1175,11 +1166,12 @@ static int blosc_c(struct thread_context* thread_context, int32_t bsize,
       desti->cspeed = (float)neblock / ctime;
       desti->filter_speed = (float) neblock / filter_time;
       dest += instr_size;
+
       continue;
     }
 
     if (!dict_training) {
-      if (cbytes == neblock) {
+      if (cbytes == 0 || cbytes == neblock) {
```

Comment:
The `cbytes == 0` should be addressed above (when you re-add it), so `cbytes == neblock` is enough here.